lib/thrift/transport/ssl.ex

```diff
@@ -25,13 +25,19 @@ defmodule Thrift.Transport.SSL do
   @spec configuration([option]) ::
           {:required | :optional, [:ssl.ssl_option()]} | nil | {:error, Exception.t()}
   def configuration(opts) do
-    case Keyword.pop(opts, :enabled, false) do
-      {true, opts} ->
-        opts
-        |> handle_configure()
-        |> handle_optional()
+    {enabled, opts} = Keyword.pop(opts, :enabled, false)
 
-      {false, _} ->
+    case enabled do
+      true ->
+        case handle_configure(opts) do
+          {:ok, opts} ->
+            handle_optional(opts)
+
+          {:error, %_exception{}} = error ->
```

Comment:
This is a little defensive as elixir patterns go because the second tuple would always match given the rest of the code. Usually wed only be so defensive at the edge of an API this defensively, which youve done below from the callback result.

Reply:
Sure that's fine.